hil/ext/switches/dellnos9.py

```diff
@@ -0,0 +1,387 @@
+        """
+        for vlan in self._get_vlans(interface):
+            self._remove_vlan_from_trunk(interface, vlan[1])
+
+    def _set_native_vlan(self, interface, vlan):
+        """ Set the native vlan of an interface.
+
+        Args:
+            interface: interface to set the native vlan to
+            vlan: vlan to set as the native vlan
+
+        Method relies on the REST API CLI which is slow
+        """
+        if not self._is_port_on(interface):
+            self._port_on(interface)
+        url = self._construct_url()
+        command = 'interface vlan ' + vlan + '\r\n untagged ' + \
+            self.interface_type + ' ' + interface
+        payload = self._make_payload(CONFIG, command)
+        self._make_request('POST', url, data=payload)
+
+    def _remove_native_vlan(self, interface):
+        """ Remove the native vlan from an interface.
+
+        Args:
+            interface: interface to remove the native vlan from.vlan
+
+        Method relies on the REST API CLI which is slow.
+        """
+        vlan = self._get_native_vlan(interface)[1]
+        url = self._construct_url()
+        command = 'interface vlan ' + vlan + '\r\n no untagged ' + \
+            self.interface_type + ' ' + interface
+        payload = self._make_payload(CONFIG, command)
+        self._make_request('POST', url, data=payload)
+
+    def _port_shutdown(self, interface):
+        """ Shuts down <interface>
+
+        Turn off portmode hybrid, disable switchport, and then shut down the
+        port. All non-default vlans must be removed before calling this.
+
+        Fast method that uses the REST API (no hackery).
+        """
+
+        url = self._construct_url(interface=interface)
+        interface = self._convert_interface(self.interface_type) + \
+            interface.replace('/', '-')
+        payload = '<interface><name>%s</name><portmode><hybrid>false' \
+                  '</hybrid></portmode><shutdown>true</shutdown>' \
+                  '</interface>' % interface
+
+        self._make_request('PUT', url, data=payload)
+
+    def _port_on(self, interface):
+        """ Turns on <interface>
+
+        Turn on port and enable hybrid portmode and switchport.
+        Fast method that uses the REST API (no hackery).
+        """
+
+        url = self._construct_url(interface=interface)
+        interface = self._convert_interface(self.interface_type) + \
+            interface.replace('/', '-')
+        payload = '<interface><name>%s</name><portmode><hybrid>true' \
+                  '</hybrid></portmode><switchport></switchport>' \
+                  '<shutdown>false</shutdown></interface>' % interface
+
+        self._make_request('PUT', url, data=payload)
+
+    def _is_port_on(self, port):
+        """ Returns a boolean that tells the status of a switchport
+        Fast method that uses the REST API (no hackery)
+        """
+
+        url = self._construct_url(interface=port, suffix='\?with-defaults')
```

Comment:
As far as I can tell this is the only place you're using the suffix argument; why not remove it and just do `+ '\?with-defaults'` here? keeps the complexity out of everything else.

Also: did you really want a backslash there?

Reply:
I wanted to leave it as it is because it might be helpful later when we do some more stuff that might need suffix.

> did you really want a backslash there?

The documentation had that, so I kept it as it is. But you can put any combination of forward and backslashes and it still works.

Reply:
OK. add a comment explaining what's going on there?

Reply:
cool. will do that.